dbt/adapters/presto/connections.py

```diff
@@ -154,13 +156,17 @@ def open(cls, connection):
                 credentials.user,
                 credentials.password,
             )
+            if credentials.http_scheme and credentials.http_scheme != "https":
+                raise dbt.exceptions.RuntimeException("http_scheme must be set to 'https' for 'ldap' method.")
             http_scheme = "https"
         elif credentials.method == 'kerberos':
             auth = prestodb.auth.KerberosAuthentication()
+            if credentials.http_scheme and credentials.http_scheme != "https":
+                raise dbt.exceptions.RuntimeException("http_scheme must be set to 'https' for 'kerberos' method.")
             http_scheme = "https"
         else:
             auth = prestodb.constants.DEFAULT_AUTH
-            http_scheme = "http"
+            http_scheme = credentials.http_scheme or "http"
```

Comment:
wow, [TIL](https://stackoverflow.com/questions/13710631/is-there-shorthand-for-returning-a-default-value-if-none-in-python)!